File: 05 SWEA/D2/Python/Prob2007.py

"""
title  : 2007. 패턴 마디의 길이 (D2)
time   : 102ms
memory : 44800kb
"""

def solution():

    T = int(input())
    for tc in range(1, T+1):

        pattern = input().rstrip()
        target = 1
        
        while target < 10:

            while pattern[0] != pattern[target]:
                target += 1

            if pattern[:target] == pattern[target:target*2]:
                print(f'#{tc} {target}')
                break

            target += 1
            
solution()

File: 05 SWEA/D2/Java/src/Prob2007.java

/**
* title  : 2007. 패턴 마디의 길이 (D2)
* time   : 101ms
* memory : 18448kb
*/

import java.io.*;
 
public class Prob2007 {
    
    public static void main(String[] args) throws IOException {
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());
        for (int tc = 1; tc <= T; tc++) {

            String pattern = br.readLine();
            int pos = 0;

            while (++pos < 10) {

                while (pattern.charAt(0) != pattern.charAt(pos)) {
                    pos++;
                }

                if (pattern.substring(0, pos).equals(pattern.substring(pos, pos * 2))) {
                    break;
                }
            }

            sb.append("#" + tc + " ").append(pos).append("\n");
        }

        System.out.println(sb);
    }
}